src/internal/m365/collection/exchange/backup_test.go

```diff
@@ -1694,6 +1751,424 @@ func (suite *CollectionPopulationSuite) TestFilterContainersAndFillCollections_r
+			data: []itemContainer{
+				{
+					container: containers[0],
+					added:     []string{"a1", "a2", "a3", "a4", "a5"},
+				},
+				{
+					container: containers[1],
+					added:     []string{"a6", "a7", "a8", "a9", "a10"},
+				},
+				{
+					container: containers[2],
+					added:     []string{"a11", "a12", "a13", "a14", "a15"},
+				},
+			},
+			expect: expected{
+				// The test isn't setup to handle partial containers so the best we can
+				// do is check that all items are expected and the item limit is hit.
+				maybeHave: []itemContainer{
+					{
+						container: containers[1],
+						added:     []string{"a6", "a7", "a8", "a9", "a10"},
+					},
+					{
+						container: containers[0],
+						added:     []string{"a1", "a2", "a3", "a4", "a5"},
+					},
+					{
+						container: containers[2],
+						added:     []string{"a11", "a12", "a13", "a14", "a15"},
+					},
+				},
+				numItems: 9,
+			},
+		},
+		{
+			name: "ExcludeContainer NoLimits",
+			limits: control.Limits{
+				MaxItems:             999,
+				MaxItemsPerContainer: 999,
+				MaxContainers:        999,
+			},
+			excludes: []string{ptr.Val(containers[1].GetId())},
+			data: []itemContainer{
+				{
+					container: containers[0],
+					added:     []string{"a1", "a2", "a3", "a4", "a5"},
+				},
+				{
+					container: containers[1],
+					added:     []string{"a6", "a7", "a8", "a9", "a10"},
+				},
+				{
+					container: containers[2],
+					added:     []string{"a11", "a12", "a13", "a14", "a15"},
+				},
+			},
+			expect: expected{
+				// The test isn't setup to handle partial containers so the best we can
+				// do is check that all items are expected and the item limit is hit.
+				maybeHave: []itemContainer{
+					{
+						container: containers[0],
+						added:     []string{"a1", "a2", "a3", "a4", "a5"},
+					},
+					{
+						container: containers[2],
+						added:     []string{"a11", "a12", "a13", "a14", "a15"},
+					},
+				},
+				numItems: 10,
+			},
+		},
+		{
+			name:       "NotPreview IgnoresLimits",
+			notPreview: true,
+			limits: control.Limits{
+				MaxItems:             1,
+				MaxItemsPerContainer: 1,
+				MaxContainers:        1,
+			},
+			excludes: []string{ptr.Val(containers[1].GetId())},
+			data: []itemContainer{
+				{
+					container: containers[0],
+					added:     []string{"a1", "a2", "a3", "a4", "a5"},
+				},
+				{
+					container: containers[1],
+					added:     []string{"a6", "a7", "a8", "a9", "a10"},
+				},
+				{
+					container: containers[2],
+					added:     []string{"a11", "a12", "a13", "a14", "a15"},
+				},
+			},
+			expect: expected{
+				mustHave: []itemContainer{
+					{
+						container: containers[0],
+						added:     []string{"a1", "a2", "a3", "a4", "a5"},
+					},
+					{
+						container: containers[1],
+						added:     []string{"a6", "a7", "a8", "a9", "a10"},
+					},
+					{
+						container: containers[2],
+						added:     []string{"a11", "a12", "a13", "a14", "a15"},
+					},
+				},
+				numItems: 15,
+			},
```

Comment:
Should we add a test case for default limits i.e. where none of the limits are set, but the preview flag is set?

Reply:
I've been debating about that. Let me see if I can find a way to write the test that would make it easy to maintain and automated to set up. Otherwise if we end up with large default limits (these are sort of numbers out of a hat at the moment) we may need a lot of manual intervention to keep the test passing